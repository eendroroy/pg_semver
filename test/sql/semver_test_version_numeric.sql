LOAD 'pg_semver';

SELECT PG_SEMVER_TO_INT('0.0.0') AS v_to_i;
SELECT PG_SEMVER_TO_INT('0.0.1') AS v_to_i;
SELECT PG_SEMVER_TO_INT('0.0.2') AS v_to_i;

SELECT '0.0.0'::SEMVER::INT AS v_to_i;
SELECT '0.0.1'::SEMVER::INT AS v_to_i;
SELECT '0.0.2'::SEMVER::INT AS v_to_i;

CREATE TABLE versions(version SEMVER);

INSERT INTO versions VALUES ('0.0.0'), ('0.0.1'), ('0.0.2'), ('0.0.3'), ('1.2.3'), ('1.0.0'), ('1.0.0-alpha.0'),
                            ('1.0.0-alpha.1'), ('1.0.0-alpha.2'), ('1.0.0-beta.0'), ('1.0.0-pre.1'), ('1.0.0-rc.1');

SELECT version, version::INT AS v_to_i FROM versions;

DROP TABLE IF EXISTS versions;
