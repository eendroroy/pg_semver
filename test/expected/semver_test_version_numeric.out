LOAD 'pg_semver';
SELECT PG_SEMVER_TO_INT('0.0.0') AS v_to_i;
 v_to_i 
--------
      0
(1 row)

SELECT PG_SEMVER_TO_INT('0.0.1') AS v_to_i;
 v_to_i 
--------
      1
(1 row)

SELECT PG_SEMVER_TO_INT('0.0.2') AS v_to_i;
 v_to_i 
--------
      2
(1 row)

SELECT '0.0.0'::SEMVER::INT AS v_to_i;
 v_to_i 
--------
      0
(1 row)

SELECT '0.0.1'::SEMVER::INT AS v_to_i;
 v_to_i 
--------
      1
(1 row)

SELECT '0.0.2'::SEMVER::INT AS v_to_i;
 v_to_i 
--------
      2
(1 row)

CREATE TABLE versions(version SEMVER);
INSERT INTO versions VALUES ('0.0.0'), ('0.0.1'), ('0.0.2'), ('0.0.3'), ('1.2.3'), ('1.0.0'), ('1.0.0-alpha.0'),
                            ('1.0.0-alpha.1'), ('1.0.0-alpha.2'), ('1.0.0-beta.0'), ('1.0.0-pre.1'), ('1.0.0-rc.1');
SELECT version, version::INT AS v_to_i FROM versions;
    version    | v_to_i 
---------------+--------
 0.0.0         |      0
 0.0.1         |      1
 0.0.2         |      2
 0.0.3         |      3
 1.2.3         |    123
 1.0.0         |      1
 1.0.0-alpha.0 |    613
 1.0.0-alpha.1 |    614
 1.0.0-alpha.2 |    615
 1.0.0-beta.0  |    507
 1.0.0-pre.1   |    423
 1.0.0-rc.1    |    309
(12 rows)

DROP TABLE IF EXISTS versions;
